flink-connectors/flink-connector-kafka-0.10/src/test/java/org/apache/flink/streaming/connectors/kafka/KafkaTestEnvironmentImpl.java

```diff
@@ -116,6 +120,30 @@ public String getVersion() {
 	}
 
 	@Override
+	public <K, V> Collection<ConsumerRecord<K, V>> getAllRecordsFromTopic(Properties properties, String topic, int partition) {
+		ImmutableList.Builder<ConsumerRecord<K, V>> result = ImmutableList.builder();
+		KafkaConsumer<K, V> consumer = new KafkaConsumer<>(properties);
+		consumer.assign(ImmutableList.of(new TopicPartition(topic, partition)));
+
+		while (true) {
+			boolean processedAtLeastOneRecord = false;
```

Comment:
I'm a bit confused by this flag.
The method name is `getAllRecordsFromTopic`, but it seems like we're escaping the loop once some record is fetched.